### PR TITLE
Add a parameter for the number of max connections per file

### PR DESCRIPTION
**Pull request by anandmohan777**

### CODE REVIEW

**Review by afoix — APPROVED** [review]
LGTM!

File: src/main/java/uk/ac/ebi/ega/egafuse/config/EgaFuseApplicationConfig.java [hunk 1]
```diff
@@ -62,11 +63,12 @@ public void setFileDatabaseService(boolean isTreeStructure) {
     }
 
     @Bean
-    public AsyncLoadingCache<CacheKey, byte[]> cache(@Value("${maxCache}") int MAX_CACHE_SIZE,
+    public AsyncLoadingCache<CacheKey, byte[]> cache(@Value("${maxCache}") int MAX_CACHE_SIZE, @Value("${connectionPerFile}") int cachePrefetch,
             IFileChunkDownloadService fileChunkDownloadService) {
         return  Caffeine.newBuilder()
                         .expireAfterWrite(5, TimeUnit.HOURS)
                         .maximumSize(MAX_CACHE_SIZE)
+                        .executor(Executors.newFixedThreadPool(cachePrefetch))
```
Comment (jorizci) [inline src/main/java/uk/ac/ebi/ega/egafuse/config/EgaFuseApplicationConfig.java]: Technically the size of the executor should be max connections. Remember that we can prefetch N chunks but the user can still try to access more files.

**Review by jorizci — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -61,7 +61,8 @@ In Linux the fuse layer can also be started, stopped and restarted using shell s
 Optional arguments:
 * m : mount point path, default value: /tmp/mnt `Note: Ensure that the mount point path exists`
 * cache : the maximum size of the cache, default value: 100 `Means 100 * 10 MB = 1000 MB`
-* c : connections, download using specified number of connections, default value: 4
+* c : total max connections, total number of API connections used by the application, default value: 4
```
Comment (malloryfreeberg) [inline README.md]: I suggest:
`c : connections, maximum number of API connections used by the application, default value: 4`

File: README.md [hunk 3]
```diff
@@ -61,7 +61,8 @@ In Linux the fuse layer can also be started, stopped and restarted using shell s
...
+* cpf : connections per file, download a file using only specified number of connections, default value: 2, `preferably, c (total max connections)  >= cpf (connections per file)`
```
Comment (malloryfreeberg) [inline README.md]: I suggest:
"cpf : connections per file, download a file using the specified number of connections, default value: 2, `preferably, c (connections)  >= cpf (connections per file)`"

**Review by jorizci — APPROVED** [review]

**Pull request merged**